src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorComponentRenameEndpoint.cs

```diff
@@ -0,0 +1,292 @@
+            {
+                return;
+            }
+
+            if (!FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                return;
+            }
+
+            var uri = new UriBuilder
+            {
+                Path = documentPath,
+                Host = string.Empty,
+                Scheme = Uri.UriSchemeFile,
+            }.Uri;
+            AddEditsForCodeDocument(documentChanges, originTagHelperBinding, newName, uri, codeDocument);
+        }
+
+        public void AddEditsForCodeDocument(List<WorkspaceEditDocumentChange> documentChanges, TagHelperBinding originTagHelperBinding, string newName, Uri uri, RazorCodeDocument codeDocument)
+        {
+            var documentIdentifier = new VersionedTextDocumentIdentifier { Uri = uri };
+            foreach (var node in codeDocument.GetSyntaxTree().Root.DescendantNodes().Where(n => n.Kind == SyntaxKind.MarkupTagHelperElement))
+            {
+                if (node is MarkupTagHelperElementSyntax tagHelperElement && BindingsMatch(originTagHelperBinding, tagHelperElement.TagHelperInfo.BindingResult))
+                {
+                    documentChanges.Add(new WorkspaceEditDocumentChange(new TextDocumentEdit
+                    {
+                        TextDocument = documentIdentifier,
+                        Edits = CreateEditsForMarkupTagHelperElement(tagHelperElement, codeDocument, newName)
+                    }));
+                }
+            }
+        }
+
+        public List<TextEdit> CreateEditsForMarkupTagHelperElement(MarkupTagHelperElementSyntax element, RazorCodeDocument codeDocument, string newName)
+        {
+            var edits = new List<TextEdit>
+            {
+                new TextEdit()
+                {
+                    Range = element.StartTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                },
+            };
+            if (element.EndTag != null)
+            {
+                edits.Add(new TextEdit()
+                {
+                    Range = element.EndTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                });
+            }
+            return edits;
+        }
+
+        private static bool BindingsMatch(TagHelperBinding left, TagHelperBinding right)
+        {
+            foreach (var leftDescriptor in left.Descriptors)
+            {
+                foreach (var rightDescriptor in right.Descriptors)
+                {
+                    if (leftDescriptor.Equals(rightDescriptor))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private async Task<TagHelperBinding> GetOriginTagHelperBindingAsync(DocumentSnapshot documentSnapshot, RazorCodeDocument codeDocument, Position position)
+        {
+            var sourceText = await documentSnapshot.GetTextAsync().ConfigureAwait(false);
```

Comment:
Is there a particular reason to set `ConfigureAwait(false)` here? We have to do that in `Microsoft.VisualStudio.*` projects because of concerns around the UI thread, but I don't think we need to do this in `Microsoft.AspNetCore.Razor.LanguageServer`. I noticed that it's that way in a lot of the other code actions too, we might follow up with that (if @NTaylorMullen or @ajaybhargavb doesn't show up to tell me I'm wrong about this) to remove those instances to avoid confusion.

Reply:
By doing `ConfigureAwait(false)` it's essentially allowing the async execution to continue on any thread rather then spend extra time / cycles in re-delegating to the original.

tl;dr it's more performant 😄